src/merkle/smt/mod.rs

```diff
@@ -49,7 +74,7 @@ pub const SMT_MAX_DEPTH: u8 = 64;
 /// [SparseMerkleTree] currently doesn't support optimizations that compress Merkle proofs.
 pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
     /// The type for a key
-    type Key: Clone + Ord;
+    type Key: Clone + KeyConstrains;
```

Comment:
I agree that it could be simpler to just list all needed features for both of cases (for hash-maps and btree-maps), but I tried to not introduce any breaking changes for possible code which doesn't use `smt_hashmaps` feature.

Reply:
Since this goes into `next`, a breaking change is fine as we'll release it as a v0.14.0.

Reply:
Okay, but I still have concern, that this will require from users of SMT implementing of `Hash` for their keys even if they want to use just btree-maps. Is it worth it, what do you think?

Reply:
I don't see a bit downside to this because externally we expose SMT implementations with pre-defined keys (e.g., either words to u64 values).